Components/Sensors/LS7366R.cs

```diff
@@ -0,0 +1,258 @@
+            IndexConfig = IndexConfig.DISABLE,
+
+            SynchronousIndex = false,
+            DivideClkBy2 = false,
+
+            // MDR1
+            CounterMode = CounterMode.BYTE_4,
+
+            CountEnable = true,
+
+            FlagOnIDX = false,
+            FlagOnCMP = false,
+            FlagOnBW = false,
+            FlagOnCY = false,
+        };
+
+
+        /// <summary> Initializes the LS7366R SPI encoder counter chip. </summary>
+        /// <param name="SPIBus"> SPI Bus to communicate with </param>
+        /// <param name="ChipSelect"> Chip Select to use with SPI </param>
+        /// <param name="CountEnable"> Digital Out to enable the counters </param>
+        public LS7366R(ISPIBus SPIBus, IDigitalOut ChipSelect, IDigitalOut CountEnable = null)
+        {
+            this.ChipSelect = ChipSelect;
+            this.CountEnable = CountEnable;
+            this.SPIBus = SPIBus;
+        }
+
+        /// <summary> Configures the device given a configuration </summary>
+        /// <param name="Configuration"> Configuration to use. </param>
+        public void Configure(Configuration Configuration)
+        {
+            // Bools to bytes
+            byte SynIndex = Configuration.SynchronousIndex ? (byte)1 : (byte)0;
+            byte FltrClk = Configuration.DivideClkBy2 ? (byte)1 : (byte)0;
+
+            byte CntEnable = Configuration.CountEnable ? (byte)0 : (byte)1;
+
+            byte IDXFlg = Configuration.FlagOnIDX ? (byte)1 : (byte)0;
+            byte CMPFlg = Configuration.FlagOnCMP ? (byte)1 : (byte)0;
+            byte BWFlg = Configuration.FlagOnBW ? (byte)1 : (byte)0;
+            byte CYFlg = Configuration.FlagOnCY ? (byte)1 : (byte)0;
+
+            // Setup MDR0 byte
+            byte MDR0 = (byte)Configuration.QuadMode;
+            MDR0 |= (byte)((int)Configuration.CountMode << 2);
+            MDR0 |= (byte)((int)Configuration.IndexConfig << 4);
+            MDR0 |= (byte)(SynIndex << 6);
+            MDR0 |= (byte)(FltrClk << 7);
+
+            // Setup MDR1 byte
+            byte MDR1 = (byte)Configuration.CounterMode;
+            MDR1 |= (byte)(CntEnable << 2);
+            MDR1 |= (byte)(IDXFlg << 4);
+            MDR1 |= (byte)(CMPFlg << 5);
+            MDR1 |= (byte)(BWFlg << 6);
+            MDR1 |= (byte)(CYFlg << 7);
+
+            // Clear MDR0 to zero
+            SPIBus.Write(ChipSelect, new byte[] { 0b00_001_000 }, 1);
+            // Clear MDR1 to zero
+            SPIBus.Write(ChipSelect, new byte[] { 0b00_010_000 }, 1);
+            // Write MDR0
+            SPIBus.Write(ChipSelect, new byte[] { 0b10_001_000, MDR0 }, 2);
+            // Write MDR1
+            SPIBus.Write(ChipSelect, new byte[] { 0b10_010_000, MDR1 }, 2);
+            // Clear CNT to zero
+            SPIBus.Write(ChipSelect, new byte[] { 0b00_100_000 }, 1);
+        }
+
+        /// <summary> Configures device with a default configuration </summary>
+        public void Configure()
```

Comment:
This could be a one-liner, but it's not worth the effort to change unless there is an actual issue...